### PR TITLE
fix(tiptap-editor): delete link from editor

### LINKED ISSUE #698: Delete link not working
Whenever I try to put up a link and try to delete it  it doesn't works.

https://github.com/user-attachments/assets/d16f914a-6097-4e11-8eec-5491791e5895

### PR DESCRIPTION
**Pull request by SySagar**
## What does this PR do?

Fixes #698

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [x] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Tested my code in a local environment
- [x] Commented on my code in hard-to-understand areas
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

## Additonal note:
I have also added custom plugin so that on hovering over editor links it would show different cursor.

### CODE REVIEW

File: apps/web/tailwind.config.ts [hunk 1]
```diff
@@ -14,6 +14,14 @@ const modifierPlugin = plugin((_) => {
...
 });
 
+const linkPlugin = plugin(function ({ addUtilities }) {
+  addUtilities({
+    '.editor-content a': {
+      cursor: 'pointer'
+    }
+  });
+});
+
```
Comment (BlankParticle) [inline apps/web/tailwind.config.ts:24]: don't do this

File: apps/web/src/components/editor/index.tsx [hunk 2]
```diff
@@ -59,7 +59,7 @@ export const Editor = memo(
       <div className="overflow-y-auto">
         <EditorRoot>
           <EditorContent
-            className="h-full w-full p-1"
+            className="editor-content h-full w-full p-1"
```
Comment (BlankParticle) [inline apps/web/src/components/editor/index.tsx:62]: if you want underline in links edit the editrProps classlist and add `prose-a:underline` there

**Review by BlankParticle — CHANGES_REQUESTED** [review]
~

File: apps/web/src/components/editor/index.tsx (hunk 2 above)
Comment (SySagar) [reply apps/web/src/components/editor/index.tsx:62]: works. Will do that asap

Comment (SySagar) [reply apps/web/src/components/editor/index.tsx:62]: However I wanted cursor style change on hovering over links.

File: apps/web/src/components/editor/index.tsx [hunk 3]
```diff
@@ -59,7 +59,7 @@ export const Editor = memo(
       <div className="overflow-y-auto">
         <EditorRoot>
           <EditorContent
-            className="h-full w-full p-1"
+            className="prose-a:cursor-pointer h-full w-full p-1"
```
Comment (BlankParticle) [inline apps/web/src/components/editor/index.tsx:62]: don't put the class here
put it while other `prose-` classes are in `editorProps`

**Review by BlankParticle — CHANGES_REQUESTED** [review]
Put the class at right place

File: apps/web/src/components/editor/index.tsx (hunk 2 above)
Comment (SySagar) [reply apps/web/src/components/editor/index.tsx]: Capture

You mean here right?

**Review by BlankParticle — APPROVED** [review]

**Pull request merged**